explore/src/main/scala/explore/targeteditor/ElevationPlotSemester.scala

```diff
@@ -57,111 +58,108 @@ object ElevationPlotSemester:
       .useContext(AppContext.ctx)
       .useResizeDetector()
       .useSerialState(none[Chart_])
-      .useEffectWithDepsBy((props, _, _, chartOpt) => chartOpt.value.isDefined)(
-        (props, ctx, _, chartOpt) =>
-          _ =>
-            chartOpt.value.value
-              .map { chart =>
-                import ctx.given
+      .useEffectWithDepsBy((props, _, _, chartOpt) =>
+        (props.options.semester, props.options.site, props.coords.value, chartOpt.value.isDefined)
+      )((_, ctx, _, chartOpt) =>
+        (semester, site, coords, _) =>
+          chartOpt.value.value
+            .map { chart =>
+              import ctx.given
 
-                val series = chart.series(0)
-                val xAxis  = chart.xAxis(0)
+              val series = chart.series(0)
+              val xAxis  = chart.xAxis(0)
 
```

Comment:
All right, this is fun! I'm still getting the errors if I do a browser reload directly on a page with the semester plot, such as `https://local.lucuma.xyz:8080/p-103/targets/obs/o-213/target/t-505` or `https://local.lucuma.xyz:8080/p-103/observation/o-213/target/t-505`. However, if I then switch sites or observations or anything, it works from then on. I put some `println`s here, and it turns out that under circumstances where we get the errors, we have times where `chartOpt` is defined, but `chart.series` and `chart.xAxis` are NOT. 

It really seems like this function and the render function are called way too many times when loading the page OR just changing the site or something. Since the `chart` in `chartOpt` is mutable, maybe we're having problems with the repeated calls?

I switched back to `react 17` and the problem on initial load went away. However, without the new `withKey` in `ElevationPlotSection`, the problem with switching sites, etc. still persisted. In this case, however, there were never cases where `series` and `xAxis` were undefined here, so it is a slightly different problem. But, it did work when changing observations. Even with react 17 this method and render are called dozens of times on load and many times when switching sites, etc.

In summary, `react 17` plus the change in ElevationPlotSection completely fixes the problem. So, I'm guessing that another `withKey` might be needed somewhere for react 18? But, if we have to keep chasing `withKey`s for react 18, we could be in for a long ride.

Reply:
Well, I just went back to react 17 to test something again, and this time it did NOT fix the error on load. 🤯 

However, if I switch to react 17, or just go back to `renderIntoDom` in Explore.scala, I have been unable to reproduce the "double click on target table" issue that Carlos is investigating.